README.md

```diff
@@ -51,6 +51,53 @@ be arbitrarily chosen.
+
+To quantifiably measure a membership inference attack, two metrics are used:
+
+
+
+*   ‘Precision’ describes the fraction of records inferred as members that are indeed members of the training set.
+*   ‘Recall’ describes the fraction of the training dataset’s members that are correctly inferred as members of the training set by an attacker.
+
+All three attacks are generic and not based on any particular dataset or model type.
+
+#### Membership Inference Back-Box
+
+This is the implementation of the original idea of a black-box membership inference attack as provided by [Shokri et al.]([https://arxiv.org/abs/1610.05820](https://arxiv.org/abs/1610.05820)). It aims to abuse the abnormal confidence of a target model when queried on their training data to differentiate between members and non-members of a model’s training set. The confidence is a vector of probabilities, the confidence values, one value per class, estimating the likelihood of a given input data record belonging to a certain class.
+
+Provided a model, the target model (trained on an unknown labeled set of data _(X,y)<sub>target</sub>_), and a separate labeled set of data _(X,y)<sub>attack</sub>_, preferably most similar to the one that the target model was trained on, this attack builds a binary meta-classifier using _(X,y)<sub>attack</sub>_, which decides whether or not a data point was part of the target model’s training set or not. 
+
+It does so by training a multitude of so-called shadow models on (disjunct and possibly overlapping) partitions of _(X,y)<sub>attack</sub>_, each imitating the behaviour of the target model, whilst knowing the “ground truth” for each shadow model, i.e. which data records their training set partitions consist of. Using the shadow models to classify a data point that was not part of _(X,y)<sub>attack</sub>_, and comparing the “unseen data confidence” of the shadow models with the one of the target model, it is able to recognize the occurrence of abnormal confidence values in the target model, indicating when a data point was part of the data that the target model was trained on.
+
+Once the meta-classifier is trained, it can be used to infer membership status on an array of one or more unknown unlabeled data records _X<sub>unkown</sub>_, returning for each record either  a 1, indicating a member, or a 0, indicating a non-member of the target model’s training set _(X,y)<sub>target</sub>_.
+
+The current implementation gives the user the choice of the attack model meta-classifier. The default is a neural network, but random forests and gradient boosting models can be chosen as well.
+
+![](docs/mia_blackbox.png)
```

Comment:
this image does have a transparent background but the others have not. It is only noticable in Github dark mode, but maybe a small change

Reply:
I've changed all images to have white background and also higher resolution.